Migrate preferences menu to react

dkacperski97 opened the pull request:
**Changes**
Updates the User Preferences menu to be React when using the experimental layout. I didn't migrate admin and user sections, because they weren't displayed on this page for the experimental layout anyway (they are now located in the AppUserMenu). I'm not sure if ripple should be disabled for buttons, but that how it is on the legacy page.

![image](https://github.com/jellyfin/jellyfin-web/assets/21142851/a52e1a6b-950a-4543-bdde-830f29fbd1d0)

grhallenbeck commented on src/apps/experimental/routes/user/menu/index.tsx:
```diff
@@ -0,0 +1,141 @@
+                        <ListItemMui disablePadding divider>
+                            <ListItemButton
+                                component='a'
+                                href={'#/mypreferencesdisplay.html?userId=' + userId}
```
For string building like this, it's generally a better practice to use template literals like `` `#/mypreferencesdisplay.html?userId=${userId}` ``

dkacperski97 replied:
I didn't see the need to use template literals for strings that only have one variable at the end of them, but I changed them to template literals.

grhallenbeck commented on src/apps/experimental/routes/user/menu/index.tsx:
```diff
@@ -0,0 +1,141 @@
+            title={globalize.translate('Settings')}
+        >
+            <div className='padded-left padded-right padded-bottom-page padded-top'>
+                <div className='readOnlyContent' style={{ margin: '0 auto' }}>
```
This `readOnlyContent` doesn't seem to actually have anything to do with read only content, and just sets some responsive max widths. I wonder if we could just replace this `div` and class name with a [Box](https://mui.com/system/react-box/) component.

dkacperski97 replied:
I don't think that the Box component would help here. By default it's just div that doesn't have any breakpoints, so we would still need to define max-width somehow, rather via a className, since the `readOnlyContent` class is used in several components.

grafixeyehero commented on src/apps/experimental/routes/user/menu/hooks/usePreferencesMenuUser.tsx:
@dkacperski97 it's a good place to replace the legacy Api to the new SDK?

dkacperski97 commented:
Sorry for the mess, by mistake I force pushed the wrong branch to mine.

After fixing that, I rebased my changes on top of the current master, I replaced the legacy api with the new sdk in the usePreferencesMenuUser hook, and I added type keyword to two imports.
